tests/test_io.py

```diff
@@ -1037,12 +1040,120 @@ def test_diagnostics_data_cancelled_simulation_file_stream(self):
+            os.remove(expected_debug_file)
+
+        model.simulate(options = opts)
+
+        nose.tools.assert_false(os.path.isfile(expected_debug_file),
+                                "Test failed, found file '{}'".format(expected_debug_file))
+
+    @testattr(stddist = True)
+    def test_debug_file_not_generated_me2(self):
+        """ Verify that the debug file is not generated by enabling logging (ME2). """
+        model = self._get_bouncing_ball_dummy()
+        opts = model.simulate_options()
+        opts["logging"] = True
+
+        expected_debug_file = "{}_debug.txt".format(model.get_identifier())
+        if os.path.isfile(expected_debug_file):
+            os.remove(expected_debug_file)
+
+        model.simulate(options = opts)
+
+        nose.tools.assert_false(os.path.isfile(expected_debug_file),
+                                "Test failed, found file '{}'".format(expected_debug_file))
+
+    @testattr(stddist = True)
+    def test_debug_file_opened_in_write_mode_me1(self):
+        """ Verify that the debug file is opened in write mode if it already did exist (ME1). """
+        model = self._get_bouncing_ball_dummy(fmu_type='me1')
+        opts = model.simulate_options()
+        opts["logging"] = True
+        opts["result_handling"] = "csv" # set to anything except 'binary'
+
+        expected_debug_file = "{}_debug.txt".format(model.get_identifier())
+        if os.path.isfile(expected_debug_file):
+            os.remove(expected_debug_file)
+
+        test_str = 'thislinewillberemoved'
+        # Now add test_str to file, simulate, and verify that test_str is gone
+        with open(expected_debug_file, 'w') as f:
+            f.write(test_str)
+        model.simulate(options = opts)
+
+        # Verify
+        with open(expected_debug_file, 'r') as f:
+            l = f.readline()
+        nose.tools.assert_false(test_str in l, "Test failed, found '{}' in '{}'".format(test_str, l))
+
+
+    @testattr(stddist = True)
+    def test_debug_file_opened_in_write_mode_me2(self):
+        """ Verify that the debug file is opened in write mode if it already did exist (ME2). """
+        model = self._get_bouncing_ball_dummy()
+        opts = model.simulate_options()
+        opts["logging"] = True
+        opts["result_handling"] = "csv" # set to anything except 'binary'
+
+        expected_debug_file = "{}_debug.txt".format(model.get_identifier())
+        if os.path.isfile(expected_debug_file):
+            os.remove(expected_debug_file)
+
+        test_str = 'thislinewillberemoved'
+        # Now add test_str to file, simulate, and verify that test_str is gone
+        with open(expected_debug_file, 'w') as f:
+            f.write(test_str)
+        model.simulate(options = opts)
+
+        # Verify
+        with open(expected_debug_file, 'r') as f:
+            l = f.readline()
+        nose.tools.assert_false(test_str in l, "Test failed, found '{}' in '{}'".format(test_str, l))
+
```

Comment:
Cant't the me1 and me2 tests be merged to 1 with an option instead? Much code duplication here.